core/src/mast/mod.rs

```diff
@@ -277,6 +277,35 @@ impl MastForest {
     }
 }
 
+/// Iterates over all the nodes a root depends on, in pre-order.
+/// The iteration can include other roots in the same forest.
+pub struct RootIterator<'a> {
```

Comment:
```suggestion
pub struct SubtreeIterator<'a> {
```

nit: I read `RootIterator` as "iterating over the roots"

Reply:
done